src/metpy/calc/thermo.py

```diff
@@ -442,6 +442,109 @@ def _lcl_iter(p, p0, w, t):
     return lcl_p, globals()['dewpoint']._nounit(vapor_pressure._nounit(lcl_p, w))
 
 
+@exporter.export
+@preprocess_and_wrap()
+@check_units('[pressure]', '[temperature]', '[temperature]')
+def ccl(pressure, temperature, dewpoint, height=None, mixed_layer_depth=None, which='top'):
+    r"""Calculate the convective condensation level (CCL).
+
+    This function is implemented by simplifying the mixing ratio and
+    the saturation vapor pressure equation to
+
+    .. math:: A(P) = ln(\frac{rP}{6.112(r+\epsilon)}) = \frac{17.67T}{T+243.5}
+
+    where :math:`P` is pressure, and :math:`T` is temperature in degrees C.
+
+    This gives us a direct relationship of how temperature changes with pressure
+    in the atmosphere with a constant mixing ratio.
```

Comment:
Since we can implement this using other functions in MetPy (see below), we can replace the equation with just a citation to the USAF manual:
```suggestion
    This function is implemented directly based on the definition of the CCL, as in [USAF1990]_, and finding
    where the ambient temperature profile intersects the line of constant mixing ratio starting at the surface,
    using the surface dewpoint.
```